4.4 does not compile in HPUX

|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.4.0 | Wazuh component | Agent | Sources | HPUX |

<!--
Whenever possible, issues should be created for bug reporting and feature requests.
For questions related to the user experience, please refer:
- Wazuh mailing list: https://groups.google.com/forum/#!forum/wazuh
- Join Wazuh on Slack: https://wazuh.com/community/join-us-on-slack

Please fill the table above. Feel free to extend it at your convenience.
-->

<!--

You may want to set debug options `<component>.debug=2` (see https://documentation.wazuh.com/current/user-manual/reference/internal-options.html) to get verbose logs. This may help investigate the issue.

-->

We have found that the 4.4 branch does not compile properly in HPUX.

When we try to run a source installation in HPUX systems we obtain the following messages:
```
13:08:00      CC client-agent/notify.o
13:08:00  <command-line>: warning: ISO C99 requires whitespace after the macro name
13:08:00  In file included from ./headers/enrollment_op.h:26,
13:08:00                   from ./headers/shared.h:302,
13:08:00                   from client-agent/notify.c:11:
13:08:00  ./headers/sec.h:57: error: expected specifier-qualifier-list before ‘_Atomic’
13:08:00  gmake[1]: *** [Makefile:1810: client-agent/notify.o] Error 1
13:08:00  gmake: *** [Makefile:754: agent] Error 2
13:08:00  grep: can't open /etc/os-release
13:08:00  install: cannot stat 'wazuh-logcollector': No such file or directory
13:08:00  install: cannot stat 'wazuh-syscheckd': No such file or directory
13:08:00  install: cannot stat 'wazuh-execd': No such file or directory
13:08:00  install: cannot stat 'manage_agents': No such file or directory
13:08:00  install: cannot stat 'wazuh-modulesd': No such file or directory
13:08:00  cp: cannot access default-firewall-drop: No such file or directory
13:08:00  install: cannot stat 'firewall-drop': No such file or directory
13:08:00  install: cannot stat 'default-firewall-drop': No such file or directory
13:08:00  install: cannot stat 'pf': No such file or directory
13:08:00  install: cannot stat 'npf': No such file or directory
13:08:00  install: cannot stat 'ipfw': No such file or directory
13:08:00  install: cannot stat 'firewalld-drop': No such file or directory
13:08:00  install: cannot stat 'disable-account': No such file or directory
13:08:00  install: cannot stat 'host-deny': No such file or directory
13:08:00  install: cannot stat 'ip-customblock': No such file or directory
13:08:00  install: cannot stat 'restart-wazuh': No such file or directory
13:08:00  install: cannot stat 'route-null': No such file or directory
13:08:00  install: cannot stat 'kaspersky': No such file or directory
13:08:00  install: cannot stat 'wazuh-slack': No such file or directory
13:08:00  install: cannot stat 'wazuh-agentd': No such file or directory
13:08:00  install: cannot stat 'agent-auth': No such file or directory

```

--- a/src/headers/shared.h
+++ b/src/headers/shared.h
@@ -99,6 +99,11 @@
 #ifdef __cplusplus
 #include <atomic>
 #define _Atomic(T) std::atomic<T>
+#else
+#ifdef hpux
+// TODO: remove this line after upgrading GCC on HP-UX
+#define _Atomic(T) T
+#endif
 #endif
 
 #include <time.h>

--- a/src/os_crypto/shared/msgs.c
+++ b/src/os_crypto/shared/msgs.c
@@ -27,14 +27,14 @@ static void ReloadCounter(const keystore *keys, unsigned int id, const char * ci
 static char *CheckSum(char *msg, size_t length) __attribute((nonnull));
 
 /* Sending counts */
-static _Atomic unsigned int global_count = 0;
-static _Atomic unsigned int local_count  = 0;
+static _Atomic (unsigned int) global_count = 0;
+static _Atomic (unsigned int) local_count  = 0;
 
 /* Average compression rates */
-static _Atomic unsigned int evt_count = 0;
-static _Atomic unsigned int rcv_count = 0;
-static _Atomic size_t c_orig_size = 0;
-static _Atomic size_t c_comp_size = 0;
+static _Atomic (unsigned int) evt_count = 0;
+static _Atomic (unsigned int) rcv_count = 0;
+static _Atomic (size_t) c_orig_size = 0;
+static _Atomic (size_t) c_comp_size = 0;
 
 /* Global variables (read from define file) */
 unsigned int _s_comp_print = 0;

